core/src/utils/trapezoid_profile.cpp

```diff
@@ -0,0 +1,122 @@
+#include "../core/include/utils/trapezoid_profile.h"
+#include "../core/include/utils/math_util.h"
+#include <cmath>
+
+/**
+ * @brief Construct a new Trapezoid Profile object
+ * 
+ * @param max_v Maximum velocity the robot can run at
+ * @param accel Maximum acceleration of the robot
+ */
+TrapezoidProfile::TrapezoidProfile(double max_v, double accel)
+: start(0), end(0), max_v(max_v), accel(accel)  {}
+
+void TrapezoidProfile::set_max_v(double max_v)
+{
+    this->max_v = max_v;
+}
+
+void TrapezoidProfile::set_accel(double accel)
+{
+    this->accel = accel;
+}
+
+void TrapezoidProfile::set_endpts(double start, double end)
+{
+    this->start = start;
+    this->end = end;
+}
+
+// Kinematic equations as macros
+#define CALC_POS(time_s,a,v,s) ((0.5*(a)*(time_s)*(time_s))+((v)*(time_s))+(s))
```

Comment:
Is there a reason why these calculations are done as macros rather than a function?